hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -1742,6 +1742,71 @@ case class TableFilterIntervals(child: TableIR, intervals: IndexedSeq[Interval],
   }
 }
 
+case class TableGroupWithinPartitions(child: TableIR, n: Int) extends TableIR {
+  lazy val children: IndexedSeq[BaseIR] = Array(child)
+
+  lazy val rowCountUpperBound: Option[Long] = child.rowCountUpperBound
+
+  override lazy val typ: TableType = TableType(
+    child.typ.keyType ++ TStruct(("grouped_fields", TArray(child.typ.rowType))),
+    child.typ.key,
+    child.typ.globalType)
+
+  def copy(newChildren: IndexedSeq[BaseIR]): TableIR = {
+    val IndexedSeq(newChild: TableIR) = newChildren
+    TableGroupWithinPartitions(newChild, n)
+  }
+
+  override def execute(ctx: ExecuteContext): TableValue = {
+    val prev = child.execute(ctx)
+    val prevRVD = prev.rvd
+    val prevRowType = prev.rvd.typ.rowType
+    val prevKeyType = prev.rvd.typ.kType
+
+    val groupedElementsPType = PArray(prevRVD.typ.rowType, true)
```

Comment:
Not sure if this should be `true` or `false`. I know that this array won't have missingness. Tried it both ways and still segfaults.